src/daemon/metric.c

```diff
@@ -29,20 +29,23 @@
 
 #include "metric.h"
 #include "plugin.h"
+#include "utils/utf8/utf8.h"
 
-/* Label names must match the regex `[a-zA-Z_][a-zA-Z0-9_]*`. Label names
- * beginning with __ are reserved for internal use.
- *
- * Source:
- * https://prometheus.io/docs/concepts/data_model/#metric-names-and-labels */
-#define VALID_LABEL_CHARS                                                      \
+/* If these characters are used in resource attribute names or metric label
+ * names, they will not cause quotes to be printed when formatting the metric
+ * name. Resource attribute values and metric label values are always printed in
+ * quotes. */
+#define UNQUOTED_LABEL_CHARS                                                   \
   "abcdefghijklmnopqrstuvwxyz"                                                 \
   "ABCDEFGHIJKLMNOPQRSTUVWXYZ"                                                 \
-  "0123456789_.-"
+  "0123456789_.-:"
 
 /* Metric names must match the regex `[a-zA-Z_:][a-zA-Z0-9_:]*` */
 // instrument-name = ALPHA 0*254 ("_" / "." / "-" / "/" / ALPHA / DIGIT)
-#define VALID_NAME_CHARS VALID_LABEL_CHARS "/"
+#define VALID_NAME_CHARS                                                       \
+  "abcdefghijklmnopqrstuvwxyz"                                                 \
+  "ABCDEFGHIJKLMNOPQRSTUVWXYZ"                                                 \
+  "0123456789_.-/"
```

Comment:
After `git grep abcdefghijklmnopqrstuvwxyz`, I wonder whether it would make sense to add `ALNUM_CHARS` define that would be used by both of these:
```
#define UNQUOTED_LABEL_CHARS  ALNUM_CHARS "_.-:"
#define VALID_NAME_CHARS      ALNUM_CHARS "_.-/"
```

And also in:
* `write_prometheus.c`: `VALID_LABEL_CHARS` / `VALID_NAME_CHARS`
* `format_stackdriver.c`: `metric_type()`::`valid_chars`

But that could be another PR.